From e9369f3ea15887207805033d40684a23d74693a2 Mon Sep 17 00:00:00 2001
From: Vasco Tenner <vasco@tenner.nl>
Date: Sat, 29 Dec 2018 17:35:20 +0100
Subject: [PATCH] Support for new style UTC timestamps issue #206

---
 src/libambit/pmem20.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/libambit/pmem20.c b/src/libambit/pmem20.c
index 6e3a928..a613d50 100644
--- a/src/libambit/pmem20.c
+++ b/src/libambit/pmem20.c
@@ -1162,12 +1162,11 @@ static void add_time(ambit_date_time_t *intime, int32_t offset, ambit_date_time_
     outtime->minute = tm->tm_min;
     outtime->hour = tm->tm_hour;
     outtime->day = tm->tm_mday;
-    outtime->month = tm->tm_mon;
-    outtime->year = tm->tm_year;
+    outtime->month = tm->tm_mon + 1;
+    outtime->year = tm->tm_year + 1900;
 }
 
 static int is_leap(unsigned int y) {
-    y += 1900;
     return (y % 4) == 0 && ((y % 100) != 0 || (y % 400) == 0);
 }
 
@@ -1180,11 +1179,11 @@ static void to_timeval(ambit_date_time_t *ambit_time, struct timeval *timeval) {
     timeval->tv_usec = 0;
     int i;
 
-    for (i = 70; i < ambit_time->year; ++i) {
+    for (i = 1970; i < ambit_time->year; ++i) {
         timeval->tv_sec += is_leap(i) ? 366 : 365;
     }
 
-    for (i = 0; i < ambit_time->month; ++i) {
+    for (i = 0; i < ambit_time->month-1; ++i) {
         timeval->tv_sec += ndays[is_leap(ambit_time->year)][i];
     }
     timeval->tv_sec += ambit_time->day - 1;
